spec/javascripts/components/story/expanded_story/expanded_story_notes_spec.js

```diff
@@ -14,141 +17,142 @@ describe('<ExpandedStoryNotes />', () => {
+    const createNoteButton = container.querySelector('.create-note-button');
+    const storySectionNotes = container.querySelector('.Story__section__notes');
+    const notelist = container.querySelector('.markdown-wrapper__text-right');
+
+    expect(createNoteText).toBeInTheDocument();
+    expect(createNoteButton).toBeInTheDocument();
+    expect(storySectionNotes).toBeInTheDocument();
+    expect(notelist).toBeInTheDocument();
   });
 
   describe('when component is enabled', () => {
     describe('when user create a new note', () => {
       it('triggers the onCreate callback passing the note', () => {
-        const story = newStory();
-
-        const change = 'newNote';
-
         const onCreateSpy = vi.fn();
+        const change = 'newNote';
+        const props = {
+          onCreate: onCreateSpy,
+          onDelete,
+          disabled: false,
+        };
 
-        const wrapper = mount(
-          <ExpandedStoryNotes
-            story={story}
-            onCreate={onCreateSpy}
-            onDelete={onDelete}
-            disabled={false}
-          />
-        );
-
-        const textArea = wrapper.find('.create-note-text');
-        const button = wrapper.find('.create-note-button input');
+        const { container } = renderComponent(props);
+        const textArea = container.querySelector('.create-note-text');
+        const button = container.querySelector('.create-note-button input');
 
-        textArea.simulate('change', { target: { value: change } });
-        button.simulate('click');
+        fireEvent.change(textArea, { target: { value: change } });
+        fireEvent.click(button);
 
         expect(onCreateSpy).toHaveBeenCalledWith(change);
+        expect(onCreateSpy).toHaveBeenCalledTimes(1);
       });
     });
 
     it('disables the add note button if text area is empty', () => {
-      const story = newStory();
-
-      const wrapper = shallow(
-        <ExpandedStoryNotes
-          story={story}
-          onCreate={onCreate}
-          onDelete={onDelete}
-          disabled={false}
-        />
-      );
-
-      const textArea = wrapper.find('.create-note-text');
-      const button = wrapper.find('.create-note-button input');
-
-      textArea.simulate('change', { target: { value: '' } });
-      expect(button.prop('disabled')).toBe(true);
+      const props = {
+        onCreate,
+        onDelete,
+        disabled: false,
+      };
+
+      const { container } = renderComponent(props);
+      const textArea = container.querySelector('.create-note-text');
+      const button = container.querySelector('.create-note-button input');
+      fireEvent.change(textArea, { target: { value: '' } });
+
+      expect(button).toBeDisabled();
     });
   });
 
   describe('when component is disabled', () => {
     it('does not render a create note button', () => {
       const notes = [{ id: 0, note: 'foo' }];
       const story = newStory(notes);
+      const props = {
```

Comment:
Can't we move this props to the above context and use it many times rather than declaring on each test describe? We could move this to the line 112 and remove it from the lines 115, 131 and 146

Reply:
Done!

[f1ca5ad](https://github.com/Codeminer42/cm42-central/pull/926/commits/f1ca5ad754903fb75e8cf9f2a22a5ea5d110713c)